### PR TITLE
M057: Integrates preloading of External DataSources as meta data resouces for use in Geonetwork

### PR DESCRIPTION
**Pull request by birdage**
adds source and tests for eoi geonetwork meta data ingestion via preload datasources

### CODE REVIEW

**Comment by birdage** [comment]
@mmeisinger @jamie-cyber1 please review

**Comment by jamie-cyber1** [comment]
Can you rebase on ooici master branch to remove the merge commits?

**Comment by birdage** [comment]
@jamie-cyber1 i rebased and fix the broken test, please review

File: ion/processes/data/registration/eoi_registration_process.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+                return True
+            else:
+                return False
+        except Exception as e:
+            #SERVICE IS REALLY NOT AVAILABLE
+            log.warn("importer service is really not available...%s", e)
+            return False    
+
+
+    def _register_data_source(self, event, *args, **kwargs):
```
Comment (mmeisinger) [inline ion/processes/data/registration/eoi_registration_process.py:64]: You should place a general try-except Exception block around the main threads. These are long running and you cannot afford to lose them to some random exception

**Comment by mmeisinger** [comment]
Looks good

**Comment by birdage** [comment]
@jamie-cyber1 can we merge? i will address Michael's comment in a subsequent pull request

**Pull request merged**